### ISSUE TITLE
Testing from API console vs command line - different behavior

### ISSUE THREAD

**Issue by pyvotal-cguers** [aws/chalice#536]
Why is it that testing an API+Lambda created with Chalice returns an error when tested from the API Gateway console, but it works (mostly) when tested from the command line (via `httpie`)? (I get a timeout from the API Gateway at the command line, but the function does complete all the steps successfully; it just says it timeout, but I will create a separate issue on this timeout)

Test at API Gw Console:
```
Fri Sep 15 21:29:10 UTC 2017 : Sending request to https://lambda.us-east-2.amazonaws.com/2015-03-31/functions/arn:aws:lambda:us-east-2:1111111111111:function:myfunc-dev/invocations
Fri Sep 15 21:29:10 UTC 2017 : Received response. Integration latency: 64 ms
Fri Sep 15 21:29:10 UTC 2017 : Endpoint response body before transformations: {"body": "{\"Message\": \"BadRequestError: newEC2: Missing execution parameters\", \"Code\": \"BadRequestError\"}", "headers": {}, "statusCode": 400}
```
The _BadRequestError_ is returned by my Lambda function when there is nothing in `app.current_request.json_body` . As a matter of fact, I log the incoming `json_body` and when I run this API from the API Gw console, the Lambda function log shows `null`; not so when run from the command line.

Test via HTTPIE:
```
$ cat testData.txt | http --timeout=300 POST https://hsxxxxxx1i.execute-api.us-east-2.amazonaws.com/dev/ec2/new
HTTP/1.1 504 Gateway Timeout
Connection: keep-alive
Content-Length: 41
Content-Type: application/json
Date: Fri, 15 Sep 2017 21:24:46 GMT
Via: 1.1 a0d2f652xx4713c255448d1a4c773a5fe.cloudfront.net (CloudFront)
X-Amz-Cf-Id: azJt3alxxxHQoXZ4orbzSyTLMEv04J9jf6ZVE-HaSmQe-Pr11r1YDw==
X-Cache: Error from cloudfront
x-amzn-RequestId: 3ac9c5ab-9a5c-11e7-847e-xxxx43f53000

{
    "message": "Endpoint request timed out"
}
```

**Comment by stealthycoin** [comment]
I'm not sure I can pin down exactly what is wrong without more details about your code and the current situation. From my own experimenting it seems taht this happens when a payload goes over one of the service limits:
http://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html
http://docs.aws.amazon.com/lambda/latest/dg/limits.html

I don't know how you are testing it in the console but is it possible that the payload you provide in the console is smaller than the payload size and whatever is in `testData.txt` is larger? That would be my first guess.

**Comment by pyvotal-cguers** [comment]
No. It is exactly the same payload. I copy/paste the same json data in `testData.txt` into the API Console test, so the tests are really comparable... unless I need to do something else in the console. For example I notice there is no "Body Mapping" section in the "Integration Request", so the body is passed through as is or I am missing something.

Here is the python code snippet that checks the parameters provided via the API call body. nothing really particular to it.

```
    ...
    try:
        accountId = app.current_request.json_body['control']['accountId']
        region = app.current_request.json_body['instance']['region']
        ...
    except:
        raise BadRequestError("newEC2: Missing execution parameters")
    ...
```

Here is the `testData.txt`:

```
{
    "control": {
        "accountId"     : "123456789012"
    },
    "instance": {
        "instanceName"  : "acme-centos-test",
        "region"        : "us-east-2",
        "instance_tags"          : {
          "test_tag": "test-value"
        }
    }
}
```

**Comment by stealthycoin** [comment]
So with that command I get a 200 from `httpie` on the command line since it parses fine. Not sure how you are getting the bad request in that context since all those properties are in the JSON file. When I tried this in the API Gateway I got a 400 back which turned out it was because it doesn't set any headers, so `app.current_request.json_body` is `None` since the Content-Type was not set correctly.

I got it working fine by adding the `Content-Type:application/json` in the API Gateway console. What headers are you providing in the console?
